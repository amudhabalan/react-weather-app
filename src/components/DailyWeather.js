import React from "react";
import Grid from "@material-ui/core/Grid";
import Moment from "react-moment";

const DailyWeather = ({ weather }) => (
  <Grid item xs={2}>
    <p>
      <Moment date={weather.applicable_date} format="dddd" />
      <br></br>
      <img
        src={
          "https://www.metaweather.com/static/img/weather/ico/" +
          weather.weather_state_abbr +
          ".ico"
        }
        alt={weather.weather_state_name}
      ></img>
      <br></br>
      <span>{Math.round(weather.max_temp)}&#8451;</span> <span>{Math.round(weather.min_temp)}&#8451;</span>
      <br></br>Humidity {weather.humidity}%
    </p>
  </Grid>
);

export default DailyWeather;
