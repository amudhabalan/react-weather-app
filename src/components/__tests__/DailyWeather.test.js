import React from 'react';
import DailyWeather from '../DailyWeather';
import Adapter from 'enzyme-adapter-react-16';
import { configure, shallow } from 'enzyme';

configure({ adapter: new Adapter() });

const weather = {
  dt: 1590354000,
  sunrise: 1590330189,
  sunset: 1590382241,
  temp: {
    day: 16.98,
    min: 13.88,
    max: 16.98,
    night: 16.88,
    eve: 16.98,
    morn: 16.98
  },
  feels_like: {
    day: 14.62,
    night: 16.09,
    eve: 14.62,
    morn: 14.62
  },
  pressure: 1021,
  humidity: 93,
  dew_point: 15.84,
  wind_speed: 6.12,
  wind_deg: 221,
  weather: [
    {
      id: 804,
      main: 'Clouds',
      description: 'overcast clouds',
      icon: '04n'
    }
  ],
  clouds: 100,
  uvi: 9.92
};

describe('DailyWeather', () => {
  const wrapper = shallow(<DailyWeather weather={weather} />);
  it('should Match Snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render the correct daily max and min temperature', () => {
    expect(
      wrapper.contains(
        <React.Fragment>
          <p>
            17
            <span>&#8451;</span> max
          </p>
          <p>
            14
            <span>&#8451;</span> min
          </p>
        </React.Fragment>
      )
    ).toEqual(true);
  });

  it('should render the correct weather image', () => {
    expect(
      wrapper.contains(
        <img
          src={'http://openweathermap.org/img/wn/04n@2x.png'}
          alt="overcast clouds"
        ></img>
      )
    ).toEqual(true);
  });
});
