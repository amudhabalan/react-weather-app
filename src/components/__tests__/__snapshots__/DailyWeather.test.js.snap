// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DailyWeather should Match Snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DailyWeather
    weather={
        Object {
            "clouds": 100,
            "dew_point": 15.84,
            "dt": 1590354000,
            "feels_like": Object {
              "day": 14.62,
              "eve": 14.62,
              "morn": 14.62,
              "night": 16.09,
            },
            "humidity": 93,
            "pressure": 1021,
            "sunrise": 1590330189,
            "sunset": 1590382241,
            "temp": Object {
              "day": 16.98,
              "eve": 16.98,
              "max": 16.98,
              "min": 13.88,
              "morn": 16.98,
              "night": 16.88,
            },
            "uvi": 9.92,
            "weather": Array [
              Object {
                "description": "overcast clouds",
                "icon": "04n",
                "id": 804,
                "main": "Clouds",
              },
            ],
            "wind_deg": 221,
            "wind_speed": 6.12,
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h3>
          <t
                    ago={false}
                    calendar={false}
                    date={1590354000}
                    decimal={false}
                    element={null}
                    filter={[Function]}
                    format="ddd"
                    fromNow={false}
                    interval={60000}
                    local={false}
                    onChange={[Function]}
                    titleFormat=""
                    toNow={false}
                    unit={null}
                    unix={true}
                    utc={false}
                    withTitle={false}
          />
</h3>,
        <img
          alt="overcast clouds"
          src="http://openweathermap.org/img/wn/04n@2x.png"
/>,
        <p>
          17
          <span>
                    ℃
          </span>
           max
</p>,
        <p>
          14
          <span>
                    ℃
          </span>
           min
</p>,
      ],
      "className": "day-weather",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <t
            ago={false}
            calendar={false}
            date={1590354000}
            decimal={false}
            element={null}
            filter={[Function]}
            format="ddd"
            fromNow={false}
            interval={60000}
            local={false}
            onChange={[Function]}
            titleFormat=""
            toNow={false}
            unit={null}
            unix={true}
            utc={false}
            withTitle={false}
/>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "ago": false,
            "calendar": false,
            "date": 1590354000,
            "decimal": false,
            "element": null,
            "filter": [Function],
            "format": "ddd",
            "fromNow": false,
            "interval": 60000,
            "local": false,
            "onChange": [Function],
            "titleFormat": "",
            "toNow": false,
            "unit": null,
            "unix": true,
            "utc": false,
            "withTitle": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "overcast clouds",
          "src": "http://openweathermap.org/img/wn/04n@2x.png",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            17,
            <span>
              ℃
</span>,
            " max",
          ],
        },
        "ref": null,
        "rendered": Array [
          17,
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "℃",
            },
            "ref": null,
            "rendered": "℃",
            "type": "span",
          },
          " max",
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            14,
            <span>
              ℃
</span>,
            " min",
          ],
        },
        "ref": null,
        "rendered": Array [
          14,
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "℃",
            },
            "ref": null,
            "rendered": "℃",
            "type": "span",
          },
          " min",
        ],
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3>
            <t
                        ago={false}
                        calendar={false}
                        date={1590354000}
                        decimal={false}
                        element={null}
                        filter={[Function]}
                        format="ddd"
                        fromNow={false}
                        interval={60000}
                        local={false}
                        onChange={[Function]}
                        titleFormat=""
                        toNow={false}
                        unit={null}
                        unix={true}
                        utc={false}
                        withTitle={false}
            />
</h3>,
          <img
            alt="overcast clouds"
            src="http://openweathermap.org/img/wn/04n@2x.png"
/>,
          <p>
            17
            <span>
                        ℃
            </span>
             max
</p>,
          <p>
            14
            <span>
                        ℃
            </span>
             min
</p>,
        ],
        "className": "day-weather",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <t
              ago={false}
              calendar={false}
              date={1590354000}
              decimal={false}
              element={null}
              filter={[Function]}
              format="ddd"
              fromNow={false}
              interval={60000}
              local={false}
              onChange={[Function]}
              titleFormat=""
              toNow={false}
              unit={null}
              unix={true}
              utc={false}
              withTitle={false}
/>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "ago": false,
              "calendar": false,
              "date": 1590354000,
              "decimal": false,
              "element": null,
              "filter": [Function],
              "format": "ddd",
              "fromNow": false,
              "interval": 60000,
              "local": false,
              "onChange": [Function],
              "titleFormat": "",
              "toNow": false,
              "unit": null,
              "unix": true,
              "utc": false,
              "withTitle": false,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "overcast clouds",
            "src": "http://openweathermap.org/img/wn/04n@2x.png",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              17,
              <span>
                ℃
</span>,
              " max",
            ],
          },
          "ref": null,
          "rendered": Array [
            17,
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "℃",
              },
              "ref": null,
              "rendered": "℃",
              "type": "span",
            },
            " max",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              14,
              <span>
                ℃
</span>,
              " min",
            ],
          },
          "ref": null,
          "rendered": Array [
            14,
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "℃",
              },
              "ref": null,
              "rendered": "℃",
              "type": "span",
            },
            " min",
          ],
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
